interactions/client/get.py

```diff
@@ -0,0 +1,158 @@
+from asyncio import sleep
+from inspect import isfunction
+from logging import getLogger
+from typing import Coroutine, Iterable, List, Type, TypeVar, Union, _GenericAlias, get_args
+
+from ..api.error import LibraryException
+from ..api.http.client import HTTPClient
+from ..api.models.channel import Channel
+from ..api.models.guild import Guild
+from ..api.models.member import Member
+from ..api.models.message import Emoji
+from ..api.models.misc import Snowflake
+from ..api.models.role import Role
+from .bot import Client
+
+log = getLogger("get")
+
+_T = TypeVar("_T")
+
+__all__ = ("get",)
+
+
+def get(*args, **kwargs):
+    # sourcery no-metrics
+
+    if len(args) == 2 and any(isinstance(_, Iterable) for _ in args):
+        raise LibraryException(message="You can only use Iterables as single-argument!", code=12)
+
+    if kwargs.get("force_http", None) and kwargs.get("force_cache", None):
+        raise LibraryException(
+            message="`force_cache` and `force_http` are mutually exclusive", code=12
+        )
+
+    if len(args) == 2:
+        client: Client
+        obj: Union[Type[_T], Type[List[_T]]]
+
+        client, obj = args
+        if not isinstance(obj, type) and not isinstance(obj, _GenericAlias):
+            raise LibraryException(
+                message="The object must not be an instance of a class!", code=12
+            )
+
+        _name = f"get_{obj.__name__.lower()}"
+        if isinstance(obj, _GenericAlias):
+            _obj = get_args(obj)[0]
+            _objects: List[_obj] = []
+
+            # TODO: add cache, include getting for IDs that are `None`
+
+            if len(list(kwargs)) == 2:
+                if guild_id := kwargs.pop("guild_id", None):
+                    _guild = Guild(**await client._http.get_guild(guild_id), _client=client._http)
```

Comment:
Any `await`s need to be moved out of this function, since it's synchronous. Ideally, any logic even touching HTTP should be in its own method.

Reply:
I know, I'm working on it